test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -413,10 +491,250 @@ public Gen<KeyspaceMetadata> build()
+                    rowsPerPartitionToCache = 0;
+                    break;
+                case 3: // num values
+                    rowsPerPartitionToCache = Math.toIntExact(rnd.next(Constraint.between(1, Integer.MAX_VALUE - 1)));
+                    break;
+                default:
+                    throw new AssertionError();
+            }
+            return new CachingParams(cacheKeys, rowsPerPartitionToCache);
+        };
+    }
+
+    public enum KnownCompactionAlgo
+    {
+        SizeTiered(SizeTieredCompactionStrategy.class),
+        Leveled(LeveledCompactionStrategy.class),
+        Unified(UnifiedCompactionStrategy.class);
+        private final Class<? extends AbstractCompactionStrategy> klass;
+
+        KnownCompactionAlgo(Class<? extends AbstractCompactionStrategy> klass)
+        {
+            this.klass = klass;
+        }
+    }
+
+    public static class CompactionParamsBuilder
+    {
+        private Gen<KnownCompactionAlgo> algoGen = SourceDSL.arbitrary().enumValues(KnownCompactionAlgo.class);
+        private Gen<CompactionParams.TombstoneOption> tombstoneOptionGen = SourceDSL.arbitrary().enumValues(CompactionParams.TombstoneOption.class);
+        private Gen<Map<String, String>> sizeTieredOptions = rnd -> {
+            if (nextBoolean(rnd)) return Map.of();
+            Map<String, String> options = new HashMap<>();
+            if (nextBoolean(rnd))
+                // computes mb then converts to bytes
+                options.put(SizeTieredCompactionStrategyOptions.MIN_SSTABLE_SIZE_KEY, Long.toString(SourceDSL.longs().between(1, 100).generate(rnd) * 1024L * 1024L));
+            if (nextBoolean(rnd))
+                options.put(SizeTieredCompactionStrategyOptions.BUCKET_LOW_KEY, Double.toString(SourceDSL.doubles().between(0.1, 0.9).generate(rnd)));
+            if (nextBoolean(rnd))
+                options.put(SizeTieredCompactionStrategyOptions.BUCKET_HIGH_KEY, Double.toString(SourceDSL.doubles().between(1.1, 1.9).generate(rnd)));
+            return options;
+        };
+        private Gen<Map<String, String>> leveledOptions = rnd -> {
+            if (nextBoolean(rnd)) return Map.of();
+            Map<String, String> options = new HashMap<>();
+            if (nextBoolean(rnd))
+                options.putAll(sizeTieredOptions.generate(rnd));
+            if (nextBoolean(rnd))
+                // size in mb
+                options.put(LeveledCompactionStrategy.SSTABLE_SIZE_OPTION, SourceDSL.integers().between(1, 2_000).generate(rnd).toString());
+            if (nextBoolean(rnd))
+                options.put(LeveledCompactionStrategy.LEVEL_FANOUT_SIZE_OPTION, SourceDSL.integers().between(1, 100).generate(rnd).toString());
+            if (nextBoolean(rnd))
+                options.put(LeveledCompactionStrategy.SINGLE_SSTABLE_UPLEVEL_OPTION, nextBoolean(rnd).toString());
+            return options;
+        };
+        private Gen<Map<String, String>> unifiedOptions = rnd -> {
```

Comment:
i don't include every config, this one class took way more time than everything else put together...